README.md

```diff
@@ -2,52 +2,93 @@
+3.  Navigate to `Tools > Extensions`.
+4.  Click `Add` and select the downloaded `.oxt` file.
+5.  Follow the on-screen instructions to install the extension.
+
+### Backend Setup
+
+To use Localwriter, you need a backend model runner.  Options include `text-generation-webui` and `Ollama`. Choose the backend that best suits your needs. Ollama is generally easier to set up. In either of these options, you will have to download and set a model. 
+
+#### text-generation-webui
+
+*   Installation instructions can be found [here](https://github.com/oobabooga/text-generation-webui).
+*   Docker image available [here](https://github.com/Atinoda/text-generation-webui-docker).
 
-In libreoffice, Navigate to Tools, then Extensions Manager, then click `Add` and find the `localwriter.oxt` file in your filesystem. You will be directed to read the license then the extension should be installed.
+After installation and model setup:
 
-You will also need to setup `text-generation-webui`, and enable the local openai API (note that this just means in this mode `text-generation-webui` will respond in the same format as open AI, all the computation still happens on your computer!). You can use whatever model you want, but I'd recommend openchat3.5 if you don't have a particular model in mind and have only 8 gigabytes of VRAM. Once the OpenAI-style API is running on localhost port 5000, localwriter should work with no additional steps. If you are using Ollama as a backend, you need to specify the `model` you are using in the settings, and likely set the endpoint url/point to Ollama's default which is `http://127.0.0.1:11434`.
+1.  Enable the local OpenAI API (this ensures the API responds in a format similar to OpenAI).
+2.  Verify that the intended model is working (e.g., openchat3.5, suitable for 8GB VRAM setups).
+3.  Set the endpoint in Localwriter to `localhost:5000` (or the configured port).
 
-## Hotkeys
+#### Ollama
 
-Extend Selection is bound to ctrl + q by default, and Edit Selection is bound to ctrl + e by default. 
+*   Installation instructions are available [here](https://ollama.com/).
+*   Download and use a model (gemma3 isn't bad)
+*   Ensure the API is enabled.
+*   Set the endpoint in Localwriter to `localhost:11434` (or the configured port).
 
 ## Settings
 
-In the settings, you can set the maximum number of additional tokens for extend selection and the maximum additional tokens (above the number of letters in the original selection) for edit selection. You can also individually set the "system prompt" for edit selection and extend selection in settings, and this prompt will always be invisibly be appended before the selection from your document send to the language model with each of these commands. For example, if you want to use a particular writing style, you can place a sample of your writing in extend selection system prompt, along with a directive to always write in a simlar style.
+In the settings, you can configure:
 
```

Comment:
Please note that the model setting must be set to the model name in the localwriter settings for ollama to work, and add a link to this forum post showinng an example ollama setup:  https://ask.libreoffice.org/t/localwriter-0-0-5-installation-and-usage/122241/5?u=jbalis